io-sim/src/Control/Monad/IOSimPOR/Internal.hs

```diff
@@ -1110,10 +1110,31 @@ execAtomically time tid tlbl nextVid0 action0 k0 =
           go ctl' read written' writtenSeq' createdSeq' nextVid (k x)
 
       ThrowStm e ->
+        {-# SCC "execAtomically.go.ThrowStm" #-}
+        case ctl of
+          AtomicallyFrame -> do
+            -- Revert all the TVar writes
+            !_ <- traverse_ (\(SomeTVar tvar) -> revertTVar tvar) written
```

Comment:
As it was pointed out in the `IOSim` case, this is common in all the branches and thus case be put in front of the `case`.